File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/mapper/SysJobMapper.java
package com.vipamp.vipclaw.admin.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.vipamp.vipclaw.admin.entity.SysJob;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;

import java.util.List;

/**
 * 定时任务 Mapper 接口
 *
 * @author vipamp
 * @since 2026-03-16
 */
@Mapper
public interface SysJobMapper extends BaseMapper<SysJob> {

    /**
     * 查询所有运行中的任务
     */
    @Select("SELECT * FROM sys_job WHERE job_status = 1 AND active = 1")
    List<SysJob> selectRunningJobs();

    /**
     * 根据任务名称和组名查询任务
     */
    @Select("SELECT * FROM sys_job WHERE job_name = #{jobName} AND job_group = #{jobGroup} AND active = 1")
    SysJob selectByNameAndGroup(@Param("jobName") String jobName, @Param("jobGroup") String jobGroup);
}

File: harnax-entity/src/main/kotlin/com/agnetix/harnax/mapper/SysJobMapper.kt
package com.agnetix.harnax.mapper

import com.agnetix.harnax.entity.SysJob
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

/**
 * SysJob Mapper interface
 */
@Mapper
interface SysJobMapper {

    // ==================== Basic CRUD Methods ====================

    fun selectById(@Param("id") id: Long): SysJob?

    fun insert(sysjob: SysJob): Int

    fun updateById(sysjob: SysJob): Int

    fun deleteById(@Param("id") id: Long): Int

    // ==================== Custom Query Methods ====================
    fun selectJobList(
        @Param("keyword") keyword: String?,
        @Param("jobStatus") jobStatus: Int?,
        @Param("currentUsername") currentUsername: String,
    ): List<SysJob>

    /**
     * Query by job name and group
     *
     * @param jobName Job name
     * @param jobGroup Job group name
     * @return Scheduled job entity
     */
    fun selectByNameAndGroup(
        @Param("jobName") jobName: String,
        @Param("jobGroup") jobGroup: String,
    ): SysJob?

    /**
     * Query all running scheduled jobs
     *
     * @return List of running scheduled jobs
     */
    fun selectRunningJobs(): List<SysJob>

    fun updateStatus(@Param("id") id: Long, @Param("status") status: Int): Int
}
